lib/CPAN2OBS.pm

```diff
@@ -0,0 +1,853 @@
+package CPAN2OBS;
+use strict;
+use warnings;
+use 5.010;
+
+use base 'Exporter';
+our @EXPORT_OK = qw/ debug info prompt /;
+
+use Data::Dumper;
+use Term::ANSIColor qw/ colored /;
+use YAML::XS qw/ DumpFile /;
+use XML::Simple qw/ XMLin /;
+use Storable qw/ retrieve store /;
+use File::Basename qw/ basename /;
+use File::Copy qw/ copy move /;
+use File::Path qw/ remove_tree /;
+use File::Spec;
+use Parse::CPAN::Packages;
+use File::Glob qw/ bsd_glob /;
+use FindBin '$Bin';
+
+use Moo;
+use namespace::clean;
+
+has data => ( is => 'rw', coerce => \&_coerce_data );
+has cpandetails => ( is => 'ro' );
+has skip => ( is => 'ro' );
+has apiurl => ( is => 'ro' );
+has cpanmirror => ( is => 'ro' );
+has cpanspec => ( is => 'ro' );
+has project_prefix => ( is => 'ro', coerce => \&_coerce_project_prefix );
+has locked => ( is => 'rw' );
+
+sub _coerce_data {
+    my ($data) = @_;
+    unless (File::Spec->file_name_is_absolute($data)) {
+        $data = File::Spec->rel2abs($data);
+    }
+    return $data;
+}
+
+sub _coerce_project_prefix {
+    my ($prefix) = @_;
+    unless ($prefix =~ m/^[A-Za-z][A-Za-z:-]+\z/) {
```

Comment:
I always find it weird to use `^` and `\z` in a regex instead of combining them consistently with `\A` or `$`.

Reply:
ok